shenfun_elasticity/solver/elastic_problem.py

```diff
@@ -14,6 +35,70 @@ def __init__(self, N, domain, elastic_law):
         self._elastic_law = elastic_law
         self._setup_problem()
 
+    def _cleanup_boundary_conditions(self):
+        for component in range(self._dim):
+            for direction in range(self._dim):
+                # check whether the bc contains both dirichlet and neumann bcs
+                # for each side (treated differently in shenfun)
+                bc = self._processed_bcs[component][direction]
+                both_dirichlet_and_neumann = True
+                for side in ('left', 'right'):
+                    if len(bc[side]) == 0:
+                        both_dirichlet_and_neumann = False
+                        break
+                    types = [bc[side][comp][0]
+                             for comp in range(len(bc[side]))]
+                    if not all([bc_type in types for bc_type in ('D', 'N')]):
+                        both_dirichlet_and_neumann = False
+                        break
+                if both_dirichlet_and_neumann:
+                    # replace by a tuple with dirichlet conditions in the first
+                    # two entries and the neumann conditions in the last two
+                    dirichlet_bcs = bc['left'][0][1], bc['right'][0][1]
+                    neumann_bcs = bc['left'][1][1], bc['right'][1][1]
+                    self._processed_bcs[component][direction] = (
+                        *dirichlet_bcs, *neumann_bcs)
+        for component in range(self._dim):
+            for direction in range(self._dim):
+                # check for bcs that fix a single value: those can not be
+                # implemented via a dictionary
+                bc = self._processed_bcs[component][direction]
+                if not isinstance(bc, dict):
+                    continue
+                if sum([len(val) for val in bc.values()]) == 1:
+                    if len(bc['left']) > 0:
+                        single_bc = bc['left'][0]
+                        index = 0
+                    else:
+                        single_bc = bc['right'][0]
+                        index = 1
+                    new_bc_format = [None, None]
+                    # for now the boundary condition for only one side
+                    # needs to be of dirichlet type
+                    assert single_bc[0] == 'D'
+                    new_bc_format[index] = single_bc[1]
+                    self._processed_bcs[component][direction] = tuple(
+                        new_bc_format)
+            # remove empty lists
+            for component in range(self._dim):
+                for direction in range(self._dim):
+                    bc = deepcopy(self._processed_bcs[component][direction])
```

Comment:
Is this deepcopy necessary? I don't think so but I am not sure.

Reply:
Without the deepcopy we change the dictionary during the iteration which raises an error. But i just noticed that a simple copy is enough.